File: CSharp/Lab1_1/Task1/LineSegment.cs
namespace Task1;

public class LineSegment
{
    public (int x, int y) Begin;
    public (int x, int y) End;

    public LineSegment((int x, int y) begin, (int x, int y) end)
    {
        this.Begin = begin;
        this.End = end;
    }

    public double GetLength()
    {
        return Math.Round(Math.Sqrt( Math.Pow(End.x - Begin.x, 2) + Math.Pow(End.y - Begin.y, 2) ), 2);
    }

    public double GetAngleRelToOY()
    {
        double hor = Math.Abs(End.x - Begin.x);
        double ver = Math.Abs(End.y - Begin.y);
        return Math.Round(Math.Atan2(hor, ver)*(180/Math.PI), 2);
    }
}

File: Cpp/Lab1_1/LineSegment.cpp
#include "LineSegment.h"
#include <math.h>

LineSegment::LineSegment(std::pair<int, int> begin, std::pair<int, int> end) : begin(begin), end(end) {}

double LineSegment::getLength() {
    return std::sqrt( std::pow(end.first - begin.first, 2) + std::pow(end.second - begin.second, 2) );
}

double LineSegment::getAngleRelToOY() {
    double hor = std::abs(end.first - begin.first);
    double ver = std::abs(end.second - begin.second);
    return std::atan2(hor, ver) * (180/M_PI);
}
